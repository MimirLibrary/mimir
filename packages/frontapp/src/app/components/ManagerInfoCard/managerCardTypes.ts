export enum ManagerCardTypes {
  OVERDUE = 'Overdue',
  DONATES = 'Donates',
}
