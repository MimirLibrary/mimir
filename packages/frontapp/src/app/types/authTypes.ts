import { IUserPayload } from '../store/slices/userSlice';

export type TAuthResponseData = Omit<IUserPayload, 'locations'> & {
  location_id: number;
  location: string;
};
