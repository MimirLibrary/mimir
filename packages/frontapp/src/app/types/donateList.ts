interface IPerson {
  __typename?: 'Person' | undefined;
  id: string;
  username: string;
  avatar: string;
}

interface IStatus {
  __typename?: 'Status' | undefined;
  id: string;
  status: string;
  person: IPerson;
}

export interface IMaterialDonate {
  __typename?: 'Material' | undefined;
  id: string;
  title: string;
  statuses: Array<IStatus | null>;
}
