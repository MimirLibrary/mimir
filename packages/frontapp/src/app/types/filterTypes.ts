export type ParamsType = {
  [key: string]: string[];
};
