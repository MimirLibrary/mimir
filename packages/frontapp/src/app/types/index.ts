export * from './materialsTypes';
export * from './authTypes';
export * from './metadata';
export * from './overdueType';
export * from './notificationList';
