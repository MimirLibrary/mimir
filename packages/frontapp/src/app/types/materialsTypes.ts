import {
  BlockedUsers,
  Location,
  Maybe,
  Message,
  Notification,
  Scalars,
  Status,
} from '@mimir/apollo-client';

export interface IStatus {
  __typename?: 'Status';
  id: string;
  created_at: any;
  status: string;
  person?: IPerson;
}

export interface IMaterial {
  __typename?: 'Material';
  title: string;
  created_at: any;
  picture?: string | null;
  author: string;
  category: string;
  id: string;
  statuses: Array<IStatus | null>;
}
interface IPerson {
  __typename?: 'Person' | undefined;
  id: string;
  username: string;
}

export interface IMaterialsState {
  searchMaterials: Array<IMaterial | null> | null | undefined;
}
export interface IReader {
  __typename?: 'Person';
  avatar?: string | null;
  id: string;
  statuses?: Array<IStatus | null | undefined> | null;
  username: string;
}
export interface IReadersState {
  searchReaders: Array<IReader | null> | null | undefined;
}
