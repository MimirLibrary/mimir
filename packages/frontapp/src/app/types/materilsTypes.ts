export interface IStatus {
  __typename?: 'Status';
  id: string;
  created_at: any;
  status: string;
}

export interface IMaterialState {
  __typename?: 'Material';
  title: string;
  created_at: any;
  picture?: string | null;
  author: string;
  category: string;
  id: string;
  statuses: Array<IStatus | null>;
}

export interface IStateMaterials {
  searchMaterials: Array<IMaterialState | null> | null | undefined;
}
