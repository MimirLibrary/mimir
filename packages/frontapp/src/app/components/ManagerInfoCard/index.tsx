import React, { FC, useState } from 'react';
import styled from '@emotion/styled';
import { ManagerCardTypes } from './managerCardTypes';
import { colors, dimensions } from '@mimir/ui-kit';
import { t } from 'i18next';
import { RoutesTypes } from '../../../utils/routes';
import { Link } from 'react-router-dom';
import { IField, IOverdueItem } from '../../types';
import { IMaterialDonate } from '../../types/donateList';
import overdue_placeholder from '../../../assets/overdue_placeholder.png';
import donate_placeholder from '../../../assets/donate_placeholder.png';

interface IManagerInfoCard {
  type: ManagerCardTypes;
  fieldsOverdue?: Array<IOverdueItem | null>;
  fieldsDonate?: Array<IMaterialDonate | null>;
}

export const WrapperCard = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background: ${colors.bg_secondary};
  height: inherit;
  width: 100%;
  box-shadow: ${colors.shadow};
  border-radius: ${dimensions.xs_1};
  padding: ${dimensions.xl_2} ${dimensions.xl_2} ${dimensions.xl_3}
    ${dimensions.xl_2};
  @media (max-width: ${dimensions.tablet_width}) {
    flex-wrap: wrap;
    min-height: 467px;
    :first-of-type {
      margin-bottom: ${dimensions.xl_3};
    }
  }
`;

export const Title = styled.p`
  font-weight: 700;
  font-size: ${dimensions.xl_2};
  line-height: ${dimensions.xl_3};
  color: ${colors.main_black};
`;

export const FieldTitle = styled.p`
  font-weight: 600;
  font-size: ${dimensions.base};
  line-height: ${dimensions.xl};
  color: ${colors.main_black};
`;

export const Description = styled.p`
  font-weight: 300;
  font-size: ${dimensions.base};
  line-height: ${dimensions.xl_2};
  color: ${colors.main_black};
`;

export const FieldDescription = styled.p`
  display: flex;
  font-weight: 400;
  font-size: ${dimensions.base};
  line-height: ${dimensions.xl};
  color: ${colors.main_black};
`;

export const WrapperFooter = styled.div`
  order: 4;
  flex-grow: 0;
  display: flex;
  flex-direction: row;
  bottom: ${dimensions.xs_1};
  justify-content: space-between;
`;
export const OpenLink = styled(Link)`
  cursor: pointer;
  font-weight: 500;
  color: ${colors.accent_color};
  font-size: ${dimensions.base};
  line-height: ${dimensions.xl};
  width: 120px;
  text-align: right;
  text-decoration: underline;
`;

const InlineOpenLink = styled(OpenLink)`
  flex: none;
  position: absolute;
  right: 71px;
  order: 1;
  flex-grow: 0;
  text-align: center;
  @media (max-width: ${dimensions.tablet_width}) {
    position: static;
    display: block;
    text-align: center;
  }
`;

export const FieldWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  flex: none;
  order: 3;
  flex-grow: 0;
  height: auto;
  padding: ${dimensions.base} 0px ${dimensions.base} ${dimensions.base};
  gap: ${dimensions.xs_2};
  background: ${colors.bg_fields};
  border-radius: ${dimensions.xs_1};
  margin-top: ${dimensions.xs_2};
  :first-of-type {
    margin-top: 0;
  }
`;
export const InlineWrapper = styled.div`
  display: flex;
  justify-content: space-between;
`;

interface IFieldOpenLinkProps {
  secondary?: string;
}
export const FieldOpenLink = styled(OpenLink)<IFieldOpenLinkProps>`
  font-weight: 400;
  width: auto;
  margin-left: 4px;
  text-decoration: underline;
  color: ${({ secondary }) =>
    secondary ? colors.problem_red : colors.accent_color};
`;

export const ColumnWrapper = styled.div`
  display: flex;
  flex-direction: column;
  row-gap: ${dimensions.xs_2};
`;

export const StyledIcon = styled.img`
  width: ${dimensions.base_3};
  height: ${dimensions.base_3};
  border-radius: 50%;
  border: 2px solid ${colors.bg_secondary};
  position: relative;
  :nth-of-type(2) {
    left: -16px;
  }
  :nth-of-type(3) {
    left: -32px;
  }
`;
export const NotificationDescription = styled(FieldDescription)`
  position: relative;
  font-weight: 500;
  font-size: ${dimensions.sm};
  left: -16px;
  align-self: center;
`;

const WrapperForEmptyBlock = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  position: relative;

  img {
    position: relative;
    z-index: 1;
  }
`;

const WrapperCircle = styled.div`
  border-radius: 50%;
  height: 270px;
  max-width: 270px;
  width: 100%;
  background-color: #f2f6ff;
  position: absolute;
`;

const TitleEmpty = styled.h3<{ top: string }>`
  font-weight: 700;
  font-size: ${dimensions.xl};
  line-height: ${dimensions.xl_2};
  color: ${colors.main_black};
  position: relative;
  top: ${(props) => props.top};
  z-index: 1;
  text-align: center;
`;

const WrapperList = styled.div`
  display: flex;
  width: 100%;
  justify-content: start;
  height: 100%;
  flex-direction: column;
  margin-top: ${dimensions.base};
`;

const ManagerInfoCard: FC<IManagerInfoCard> = ({
  type,
  fieldsOverdue,
  fieldsDonate,
}) => {
  return (
    <>
      {fieldsOverdue && (
        <WrapperCard>
          <div>
            <Title>
              {t(`ManagerInfoCard.Title.${type}`) +
                ` — (${fieldsOverdue?.length})`}
            </Title>
            <Description>
              {t(`ManagerInfoCard.Description.${type}`)}
            </Description>
          </div>
          {!fieldsOverdue.length ? (
            <WrapperForEmptyBlock>
              <WrapperCircle />
              <img src={overdue_placeholder} alt="overdue_placeholder" />
              <TitleEmpty top="45px">
                {t('PlaceholderTitle.Overdue')}
              </TitleEmpty>
            </WrapperForEmptyBlock>
          ) : (
            <WrapperList>
              {fieldsOverdue?.slice(0, 3).map((field) => (
                <FieldWrapper key={field?.id}>
                  <FieldTitle>{field?.material.title}</FieldTitle>
                  <InlineWrapper>
                    <FieldDescription>
                      {t(`ManagerInfoCard.FieldDescription.${type}`)}
                    </FieldDescription>
                    <FieldOpenLink
                      secondary="true"
                      to={`${RoutesTypes.READERS}/${field?.person.id}`}
                    >
                      {field?.person.username}
                    </FieldOpenLink>
                  </InlineWrapper>
                </FieldWrapper>
              ))}
            </WrapperList>
          )}
          <WrapperFooter>
            <>
              <InlineWrapper>
                {fieldsOverdue?.slice(0, 3).map((field) => (
                  <StyledIcon key={field?.id} src={field?.person.avatar} />
                ))}
                {fieldsOverdue!.length > 3 ? (
                  <NotificationDescription>
                    {`+${fieldsOverdue.length - 3}` +
                      t(`ManagerInfoCard.Description.More`)}
                  </NotificationDescription>
                ) : null}
              </InlineWrapper>
            </>
            {!!fieldsOverdue.length && (
              <OpenLink to={`home/${type.toLowerCase()}`}>
                {t(`ManagerInfoCard.Link.${type}`)}
              </OpenLink>
            )}
          </WrapperFooter>
        </WrapperCard>
      )}
      {fieldsDonate && (
        <WrapperCard>
          <div>
            <Title>
              {t(`ManagerInfoCard.Title.${type}`) +
                ` — (${fieldsDonate?.length})`}
            </Title>
            <Description>
              {t(`ManagerInfoCard.Description.${type}`)}
            </Description>
          </div>
          {!fieldsDonate.length ? (
            <WrapperForEmptyBlock>
              <WrapperCircle />
              <img
                src={donate_placeholder}
                alt="donate_placeholder"
                data-testid="donate-placeholder"
              />
              <TitleEmpty top="20px">{t('PlaceholderTitle.Donate')}</TitleEmpty>
            </WrapperForEmptyBlock>
          ) : (
            <WrapperList data-testid="items-list">
              {fieldsDonate?.slice(0, 3).map((field) => (
                <FieldWrapper key={field?.id}>
                  <FieldTitle>{field?.title}</FieldTitle>
                  <InlineWrapper>
                    <FieldDescription>
                      {t(`ManagerInfoCard.FieldDescription.${type}`)}
                    </FieldDescription>
                    <FieldOpenLink
                      secondary="true"
                      to={`${RoutesTypes.READERS}/${
                        field?.statuses[field?.statuses.length - 1]?.person.id
                      }`}
                    >
                      {
                        field?.statuses[field?.statuses.length - 1]?.person
                          .username
                      }
                    </FieldOpenLink>
                  </InlineWrapper>
                </FieldWrapper>
              ))}
            </WrapperList>
          )}
          <WrapperFooter>
            <>
              <InlineWrapper>
                {fieldsDonate?.slice(0, 3).map((field) => (
                  <StyledIcon
                    key={field?.id}
                    src={
                      field?.statuses[field?.statuses.length - 1]?.person.avatar
                    }
                  />
                ))}
                {fieldsDonate.length > 3 ? (
                  <NotificationDescription>
                    {`+${fieldsDonate.length - 3}` +
                      t(`ManagerInfoCard.Description.More`)}
                  </NotificationDescription>
                ) : null}
              </InlineWrapper>
            </>
            {!!fieldsDonate.length && (
              <OpenLink to={`${RoutesTypes.DONATES_FROM_USER}`}>
                {t(`ManagerInfoCard.Link.${type}`)}
              </OpenLink>
            )}
          </WrapperFooter>
        </WrapperCard>
      )}
    </>
  );
};

export default React.memo(ManagerInfoCard);
