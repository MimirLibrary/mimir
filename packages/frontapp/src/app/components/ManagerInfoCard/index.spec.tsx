import { render } from '../../../helpers/customRender';
import ManagerInfoCard from './index';
import { ManagerCardTypes } from './managerCardTypes';
import { IMaterialDonate } from '../../types/donateList';
import { screen } from '@testing-library/react';
import donate_placeholder from '../../../assets/donate_placeholder.png';

describe('render ManagerInfo Card', () => {
  it('render with correct data with donate items', () => {
    const data: Array<IMaterialDonate> = [
      {
        id: '18',
        title: 'test',
        statuses: [
          {
            status: 'Pending',
            id: '1',
            person: {
              id: '11',
              username: 'Ivan Uglovets',
              avatar: '',
            },
          },
        ],
      },
    ];

    render(
      <ManagerInfoCard type={ManagerCardTypes.DONATES} fieldsDonate={data} />
    );

    expect(screen.getByTestId('items-list')).toBeInTheDocument();
    expect(screen.getByText('test')).toBeInTheDocument();
    expect(screen.getByText('Ivan Uglovets')).toBeInTheDocument();
  });

  it('render component with empty list', () => {
    render(
      <ManagerInfoCard type={ManagerCardTypes.DONATES} fieldsDonate={[]} />
    );

    expect(screen.getByTestId('donate-placeholder')).toHaveAttribute(
      'src',
      donate_placeholder
    );
  });

  it('render component when data is undefined', () => {
    render(
      <ManagerInfoCard
        type={ManagerCardTypes.DONATES}
        fieldsDonate={undefined}
      />
    );
    expect(screen.queryByTestId('items-list')).toBeNull();
    expect(screen.queryByTestId('donate-placeholder')).toBeNull();
  });
});
