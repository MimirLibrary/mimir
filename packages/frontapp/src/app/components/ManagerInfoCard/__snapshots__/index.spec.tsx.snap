// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ManagerInfo Card snapshot testing of manager DONATES card 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  background: #FFFFFF;
  height: inherit;
  width: 100%;
  box-shadow: 0px 10px 70px rgba(26, 30, 214, 0.08);
  border-radius: 0.625rem;
  position: relative;
  padding: 1.5rem 1.5rem 1.875rem 1.5rem;
}

@media (max-width: 48rem) {
  .emotion-0 {
    box-shadow: none;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-flex-wrap: wrap;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    min-height: 300px;
    border-radius: 0;
  }

  .emotion-0:first-of-type {
    margin-bottom: 1.875rem;
  }
}

.emotion-1 {
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 1.875rem;
  color: #333333;
}

.emotion-2 {
  font-weight: 300;
  font-size: 1rem;
  line-height: 1.5rem;
  color: #333333;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  width: 100%;
}

.emotion-3 img {
  position: relative;
  z-index: 1;
}

@media (max-width: 48rem) {
  .emotion-3 img {
    height: 100px;
  }
}

.emotion-4 {
  border-radius: 50%;
  height: 270px;
  max-width: 270px;
  width: 100%;
  background-color: #f2f6ff;
  position: absolute;
}

@media (max-width: 48rem) {
  .emotion-4 {
    height: 170px;
    max-width: 170px;
  }
}

.emotion-5 {
  font-weight: 700;
  font-size: 1.25rem;
  line-height: 1.5rem;
  color: #333333;
  position: relative;
  top: 20px;
  z-index: 1;
  text-align: center;
}

@media (max-width: 48rem) {
  .emotion-5 {
    display: none;
  }
}

.emotion-6 {
  -webkit-order: 4;
  -ms-flex-order: 4;
  order: 4;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  bottom: 0.625rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

@media (max-width: 48rem) {
  .emotion-6 {
    display: none;
  }
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

@media (max-width: 48rem) {
  .emotion-7 {
    display: block;
  }
}

<div
  class="emotion-0"
  data-testid="donateCard"
>
  <div>
    <p
      class="emotion-1"
    >
      undefined — (0)
    </p>
    <p
      class="emotion-2"
    />
  </div>
  <div
    class="emotion-3"
  >
    <div
      class="emotion-4"
    />
    <img
      alt="donate_placeholder"
      data-testid="donate-placeholder"
      src="donate_placeholder.png"
    />
    <h3
      class="emotion-5"
    />
  </div>
  <div
    class="emotion-6"
  >
    <div
      class="emotion-7"
    />
  </div>
</div>
`;

exports[`render ManagerInfo Card snapshot testing of manager NOTIFICATIONS card 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  background: #FFFFFF;
  height: inherit;
  width: 100%;
  box-shadow: 0px 10px 70px rgba(26, 30, 214, 0.08);
  border-radius: 0.625rem;
  position: relative;
  padding: 1.5rem 1.5rem 1.875rem 1.5rem;
}

@media (max-width: 48rem) {
  .emotion-0 {
    box-shadow: none;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-flex-wrap: wrap;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    min-height: 300px;
    border-radius: 0;
  }

  .emotion-0:first-of-type {
    margin-bottom: 1.875rem;
  }
}

.emotion-1 {
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 1.875rem;
  color: #333333;
}

.emotion-2 {
  display: none;
  cursor: pointer;
}

@media (max-width: 48rem) {
  .emotion-2 {
    display: block;
    position: absolute;
    right: 20px;
    top: 70px;
  }
}

.emotion-3 {
  font-weight: 300;
  font-size: 1rem;
  line-height: 1.5rem;
  color: #333333;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  width: 100%;
}

.emotion-4 img {
  position: relative;
  z-index: 1;
}

@media (max-width: 48rem) {
  .emotion-4 img {
    height: 100px;
  }
}

.emotion-5 {
  border-radius: 50%;
  height: 270px;
  max-width: 270px;
  width: 100%;
  background-color: #f2f6ff;
  position: absolute;
}

@media (max-width: 48rem) {
  .emotion-5 {
    height: 170px;
    max-width: 170px;
  }
}

.emotion-6 {
  font-weight: 700;
  font-size: 1.25rem;
  line-height: 1.5rem;
  color: #333333;
  position: relative;
  top: 20px;
  z-index: 1;
  text-align: center;
}

@media (max-width: 48rem) {
  .emotion-6 {
    display: none;
  }
}

.emotion-7 {
  -webkit-order: 4;
  -ms-flex-order: 4;
  order: 4;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  bottom: 0.625rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

@media (max-width: 48rem) {
  .emotion-7 {
    display: none;
  }
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

@media (max-width: 48rem) {
  .emotion-8 {
    display: block;
  }
}

<div
  class="emotion-0"
  data-testid="notificationCard"
>
  <div>
    <p
      class="emotion-1"
    >
      undefined — (0)
    </p>
    <div
      class="emotion-2"
    >
      <svg>
        AddBookBluePlus.svg
      </svg>
    </div>
    <p
      class="emotion-3"
    />
  </div>
  <div
    class="emotion-4"
  >
    <div
      class="emotion-5"
    />
    <img
      alt="notification_placeholder"
      data-testid="notification-placeholder"
      src="notification_placeholder.png"
    />
    <h3
      class="emotion-6"
    />
  </div>
  <div
    class="emotion-7"
  >
    <div
      class="emotion-8"
    />
  </div>
</div>
`;

exports[`render ManagerInfo Card snapshot testing of manager Overdue card 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  background: #FFFFFF;
  height: inherit;
  width: 100%;
  box-shadow: 0px 10px 70px rgba(26, 30, 214, 0.08);
  border-radius: 0.625rem;
  position: relative;
  padding: 1.5rem 1.5rem 1.875rem 1.5rem;
}

@media (max-width: 48rem) {
  .emotion-0 {
    box-shadow: none;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-flex-wrap: wrap;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    min-height: 300px;
    border-radius: 0;
  }

  .emotion-0:first-of-type {
    margin-bottom: 1.875rem;
  }
}

.emotion-1 {
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 1.875rem;
  color: #333333;
}

.emotion-2 {
  font-weight: 300;
  font-size: 1rem;
  line-height: 1.5rem;
  color: #333333;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  width: 100%;
}

.emotion-3 img {
  position: relative;
  z-index: 1;
}

@media (max-width: 48rem) {
  .emotion-3 img {
    height: 100px;
  }
}

.emotion-4 {
  border-radius: 50%;
  height: 270px;
  max-width: 270px;
  width: 100%;
  background-color: #f2f6ff;
  position: absolute;
}

@media (max-width: 48rem) {
  .emotion-4 {
    height: 170px;
    max-width: 170px;
  }
}

.emotion-5 {
  font-weight: 700;
  font-size: 1.25rem;
  line-height: 1.5rem;
  color: #333333;
  position: relative;
  top: 45px;
  z-index: 1;
  text-align: center;
}

@media (max-width: 48rem) {
  .emotion-5 {
    display: none;
  }
}

.emotion-6 {
  -webkit-order: 4;
  -ms-flex-order: 4;
  order: 4;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  bottom: 0.625rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

@media (max-width: 48rem) {
  .emotion-6 {
    display: none;
  }
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

@media (max-width: 48rem) {
  .emotion-7 {
    display: block;
  }
}

<div
  class="emotion-0"
  data-testid="overdueCard"
>
  <div>
    <p
      class="emotion-1"
    >
      undefined — (0)
    </p>
    <p
      class="emotion-2"
    />
  </div>
  <div
    class="emotion-3"
  >
    <div
      class="emotion-4"
    />
    <img
      alt="overdue_placeholder"
      data-testid="overdue-placeholder"
      src="overdue_placeholder.png"
    />
    <h3
      class="emotion-5"
    />
  </div>
  <div
    class="emotion-6"
  >
    <div
      class="emotion-7"
    />
  </div>
</div>
`;
